### DISCUSSION TITLE
Dropdown filter misbehaviour during table update

### DISCUSSION THREAD

**Discussion by underdpt** [filamentphp/filament#2844] [Help]
Hello all,

I'm not sure this is a bug, I'm experimenting an odd behaviour on dropdown filters (SelectFilter or MultiselectFilter) when working with a heavy table (lots of markup). The problem is that if you're typing on the dropdown text area to search for an item while the table is still reloading, once the reload is finished the dropdown seems to reset and the search results go away, you're shown the default items (the ones that appear when you click on the dropdown).

In this gif you can see that the user search for the option "cuatro" by writing "cuatr", and he is shown the two coincidences "cuatro" and "tres", but when the reload process ends the drowdown is reset and shows all the options (see screenshot below)

![github](https://user-images.githubusercontent.com/8122137/174620145-2d7bb815-d8b7-42fe-bb25-4453c6a6001a.gif)

Screenshot after the table finishes reloading:

![imagen](https://user-images.githubusercontent.com/8122137/174621397-351f28b8-6a32-4e48-9703-2ae93f233ee0.png)

I think that the desired behaviour is that the dropdown search doesn't reset and shows only the matching results.

**Comment by danharrin** [comment]
Can you convert this to an issue please?

Also, those columns glitching out while loading seems very wrong

**Reply by underdpt** [reply]
Done, created issue #2858
